src/operator/tensor/matrix_op-inl.h

```diff
@@ -316,6 +316,12 @@ inline bool ExpandDimShape(const nnvm::NodeAttrs& attrs,
   ret[param.axis] = 1;
   for (index_t i = param.axis+1; i < ret.ndim(); ++i) ret[i] = shp[i-1];
   SHAPE_ASSIGN_CHECK(*out_attrs, 0, ret);
+
```

Comment:
When shp.ndim()==0, param.axis must be 0 too. ret[param.axis] would be 1. Why would it fail?